exercises/tom_clayton/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,158 @@
+/**
+ * Import statements
+ */
+
+#include <iostream>
+#include <sstream>
+#include <fstream>
+#include <algorithm>
+#include <cctype>
+#include <string.h>
+#include <vector>
+#include <map>
+using namespace std;
+
+/**
+ * Function to return a string filepath from user input
+ */
+
+std::string getInputFn() {
+  std::cout << "Please enter the path of the file to be analysed" << std::endl;
+  std::string fin;
+  std::cin >> fin;
+  return fin;
+}
+
+/**
+ * Function to split up a string and return a vector of words with
+ * len <4, and converted to lower case
+ */
+
+std::vector<std::string> splitByDelim(const std::string& s, char delim) {
+  
+  std::vector<std::string> tokens;
+  std::string token;
+  std::istringstream tokenStream(s);
+  
+  while (std::getline(tokenStream, token, delim)) {
+    if(token.size() < 4) {
+      continue;
+    }
+
+    std::transform(token.begin(), token.end(), token.begin(),
+    [](unsigned char c){ return std::tolower(c);});
+    
+    tokens.push_back(token);
+    //cout << token << endl;
+  }
+
+  return tokens;
+}
+
+/**
+ * Function to replace puncuation with whitespace in a string
+ */
+
+void depunctuate(std::string& line) {
+
+  std::replace_if(line.begin() , line.end(), 
+  [] (const char& c) { return std::ispunct(c) ;},' ');
+}
+
+/**
+ * Function to read lines from an input and return a vector of
+ * depunctuated, lower case words of length>=4
+ */
+
+std::vector<std::string> inputToVector() {
+
+  std::ifstream inFile;
+  std::string fin = getInputFn();
+  inFile.open(fin, ios::in);
+  if (!inFile) {
+    std::cout << "Unable to open the file.";
+    exit(1);
+  }
+
+  std::string line;
+  std::vector<std::string> words;
+  while(std::getline(inFile, line)){
+    depunctuate(line);
+    std::vector<std::string> spLine = splitByDelim(line, ' ');
+    words.insert(words.end(), spLine.begin(), spLine.end());
+  }
+  inFile.close();
+  return words;
+}
+
+/**
+ * Function to populate a map with key=word, value=no of
+ * occurrences for an input vector of single word strings
+ */ 
+
+void fillMap(std::map<std::string, int>& wordCountMap, std::vector<std::string>& words) {
```

Comment:
Please can you add const to the 2nd parameter because I don't think it's modified. Helps readability of what the function is doing

Reply:
Done